File: Recursion/lecture 5/Fraz_recursion_5.cpp
bool isPalindromeHelper(int l,int r,string &s)
{
    if(l>=r) return true;
    if(s[l]!=s[r]) return false;
    return isPalindromeHelper(l+1,r-1,s);

}
bool isPalindrome(string &s)
{
    return isPalindromeHelper(0,s.size()-1,s);
}

File: Recursion/lecture 5/Fraz_recursion_5.java
 public static Boolean isPalindromehelper(int l,int r,String s){
//         base
        if(l>=r){
               return true;
             }
           if(s.charAt(l)!= s.charAt(r)){
               return false;
           }
           return isPalindromehelper(l+1,r-1,s);
        
    }
     static boolean isPalindrome(String s) {
           int l =0;
        int r =s.length()-1;
        return isPalindromehelper(0,s.length()-1,s);
    }
